x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/InstanceTypeService.java

```diff
@@ -27,33 +27,37 @@ private static final class Holder {
         private static final Map<InstanceType, CostEntry> costsPerDatacenter;
 
         static {
-            final Map<Object, Object> objects = new HashMap<>();
+            final Map<Object, Object> objects = new HashMap<>(1024);
             final Function<String, String> dedupString = s -> (String) objects.computeIfAbsent(s, Function.identity());
-            final Map<InstanceType, CostEntry> tmp = new HashMap<>();
-            try (
-                GZIPInputStream in = new GZIPInputStream(
-                    InstanceTypeService.class.getClassLoader().getResourceAsStream("profiling-costs.json.gz")
-                );
-                XContentParser parser = XContentType.JSON.xContent().createParser(XContentParserConfiguration.EMPTY, in)
-            ) {
-                if (parser.currentToken() == null) {
-                    parser.nextToken();
-                }
-                List<Map<String, Object>> rawData = XContentParserUtils.parseList(parser, XContentParser::map);
-                for (Map<String, Object> entry : rawData) {
-                    tmp.put(
-                        new InstanceType(
-                            dedupString.apply((String) entry.get("provider")),
-                            dedupString.apply((String) entry.get("region")),
-                            dedupString.apply((String) entry.get("instance_type"))
-                        ),
-                        (CostEntry) objects.computeIfAbsent(CostEntry.fromSource(entry), Function.identity())
-                    );
+            final Map<InstanceType, CostEntry> tmp = new HashMap<>(16 * 1024);
+
+            // All files ar expected to exist.
+            for (String provider : List.of("aws", "azure")) {
+                String name = "profiling-costs-" + provider + ".json.gz";
+                try (
+                    GZIPInputStream in = new GZIPInputStream(InstanceTypeService.class.getClassLoader().getResourceAsStream(name));
+                    XContentParser parser = XContentType.JSON.xContent().createParser(XContentParserConfiguration.EMPTY, in)
+                ) {
+                    if (parser.currentToken() == null) {
+                        parser.nextToken();
+                    }
+                    List<Map<String, Object>> rawData = XContentParserUtils.parseList(parser, XContentParser::map);
+                    for (Map<String, Object> entry : rawData) {
+                        tmp.put(
+                            new InstanceType(
+                                dedupString.apply((String) entry.get("provider")),
```

Comment:
Given that we have now provider-specific files we could get rid of this property in the individual JSON files?

Reply:
Good point, done.